### ISSUE TITLE
NLB: UDP support

### ISSUE THREAD

**Issue by zmcneilly** [aws/aws-cdk#5571]
Cloudformation supports [TCP, TCP_UDP, and UDP](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-targetgroup.html#cfn-elasticloadbalancingv2-targetgroup-protocol) protocols for Network Target Groups for Network Load Balancers. However, CDK only supports the [TCP protocol](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-target-group.ts#L60). Currently this is a blocker for us to be able to adopt CDK as we need to support TCP_UDP connections.

**Comment by gazwald** [comment]
I've also run into this issue. For anyone that needs a quick workaround you can override the Protocol on the listener and target group like so:

Assuming `listener` is defined as your NLB listener:

```
cfn_listener = listener.node.default_child
cfn_listener.add_override("Properties.Protocol", "UDP")
```
Assuming `target_group` is defined as your listeners target:

```
cfn_target_group = target_group.node.default_child
cfn_target_group.add_override("Properties.Protocol", "UDP")
```

See "Raw Overrides" in the "Escape Hatches" documentation for more details about `add_override`: https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html

**Comment by njlynch** [comment]
Sorry for the lack of action in this issue. This was resolved via #8525 and released in CDK 1.46.0.
